src/main/java/uk/ac/cam/cl/dtg/isaac/api/managers/GameManager.java

```diff
@@ -980,6 +980,52 @@ public List<GameboardItem> getNextQuestionsForFilter(final GameFilter gameFilter
     return selectionOfGameboardQuestions;
   }
 
+  /**
+   * This generates a set of five question in random.
+   *
+   * @param gameFilter - to enable search
+   * @param limit      - to provide a limit of questions
+   * @return a list of questions
+   * @throws ContentManagerException - if there is a problem accessing the content repository.
+   */
+  public List<ContentDTO> generateRandomQuestions(final GameFilter gameFilter, final int limit)
+      throws ContentManagerException {
+    List<GitContentManager.BooleanSearchClause> fieldsToMap = Lists.newArrayList();
+    fieldsToMap.add(new GitContentManager.BooleanSearchClause(
+        TYPE_FIELDNAME, uk.ac.cam.cl.dtg.segue.api.Constants.BooleanOperator.AND,
+        Collections.singletonList(QUESTION_TYPE)));
+    fieldsToMap.addAll(generateFieldToMatchForQuestionFilter(gameFilter));
+
+    List<ContentDTO> questionsToReturn = Lists.newArrayList();
+
+    while (questionsToReturn.size() < limit) {
+      var results =  this.contentManager.findByFieldNamesRandomOrder(fieldsToMap,
+          0,
+          limit,
+          null);
+
+      List<ContentDTO> generatedQuestions = results.getResults();
+
+      if (generatedQuestions.size() < limit) {
+        throw new ContentManagerException("Error in generating random questions");
+      }
+
+      for (ContentDTO question : generatedQuestions) {
+        // Only keep questions that have not been superseded or deprecated.
+        if (question instanceof IsaacQuestionPageDTO) {
+          IsaacQuestionPageDTO qp = (IsaacQuestionPageDTO) question;
+          if ((qp.getSupersededBy() != null && qp.getSupersededBy().length() != 0)
```

Comment:
`!qp.getSupersededBy().isEmpty()` is a bit more readable.

Reply:
Funnily enough, that was how I wrote it originally, and checkstyle complained about the performance. I'm happy to change back if I've got your backing.

Reply:
Do you have a link to that? I'd like to understand why. `isEmpty` should be more performant than checking the length. Not that it matters that much, as readability is more important here.